### PR TITLE
CDPS-836, CDPS-847, CDPS-849, CDPS-850: Edit Profile

### PR DESCRIPTION
**Pull request by davidmiddletonmoj**
Hair type or colour
Facial hair
Face shape
Build

### CODE REVIEW

File: integration_tests/e2e/personal/edit/radiosTests.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import { Role } from '../../../../server/data/enums/role'
```
[the remark this answers is not in the document]
Comment (davidmiddletonmoj) [reply integration_tests/e2e/personal/edit/radiosTests.ts:1]: I didn't do this initially as there were a few differences in the pages and wasn't sure what I was going to do about the eye colour one next, but yeah better to have one so I've made those changes and commented out a few things until the height/weight pages are updated.

**Review by danielburnley — APPROVED** [review]

File: server/controllers/personal/personalController.ts [hunk 2]
```diff
@@ -282,4 +287,100 @@ export default class PersonalController {
...
+            .catch(error => logger.error(error))
+
+          return res.redirect(`/prisoner/${prisonerNumber}/personal#appearance`)
+        } catch (e) {
+          errors.push({ text: 'There was an error please try again' })
+        }
+
+        req.flash('fieldValue', radioField)
+        req.flash('errors', errors)
+        return res.redirect(`/prisoner/${prisonerNumber}/personal/edit/${url}`)
```
Comment (whitfield-mj) [inline server/controllers/personal/personalController.ts:382]: Why not have this in the catch block then we don't need to instantiate an empty array and mutate it?

File: server/controllers/personal/personalController.ts [hunk 3]
```diff
@@ -282,4 +287,100 @@ export default class PersonalController {
...
+          prisonId: prisonerData.prisonId,
+          correlationId: req.id,
+          page: auditPage,
+        })
+
+        res.render('pages/edit/radios', {
+          pageTitle,
+          prisonerNumber,
+          breadcrumbPrisonerName: prisonerBannerName,
+          errors: hasLength(errors) ? errors : [],
```
Comment (whitfield-mj) [inline server/controllers/personal/personalController.ts:341]: Is this any different to just returning errors?

File: server/services/personalPageService.ts [hunk 4]
```diff
@@ -45,12 +51,23 @@ export default class PersonalPageService {
...
+  async updatePhysicalCharacteristics(
+    token: string,
+    prisonerNumber: string,
+    physicalCharacteristics: Partial<PrisonPersonPhysicalCharacteristics>,
+  ): Promise<PrisonPersonPhysicalCharacteristics> {
+    const apiClient = this.prisonPersonApiClientBuilder(token)
+    return null
+    // TODO plug in API call when available
+    return apiClient.updatePhysicalCharacteristics(prisonerNumber, physicalCharacteristics)
+  }
```
[the remark this answers is not in the document]
Comment (davidmiddletonmoj) [reply server/services/personalPageService.ts:69]: Yeah, initially I just had line 68 commented out, but eslint complained about unused variables 🤷‍♂️

**Pull request merged**